### ISSUE TITLE
Oscillator frequency not set, leading to wrong clock dividers

### ISSUE THREAD

**Issue by maxgerhardt** [darkautism/pio-lgt8fx#5]
Per https://community.platformio.org/t/serial-monitor-outputting-non-recognized-characters/35833

When using a `platformio.ini` of

```ini
[env:LGT8F328P]
platform = lgt8f
board = LGT8F328P
framework = arduino
```

The board definition says it would run the CPU at 32MHz

https://github.com/darkautism/pio-lgt8fx/blob/7d5d21c74e5cd3d7734ef398d8b3c074333c718e/boards/LGT8F328P.json#L1-L5

And the builder script sets `F_CPU` accordingly

https://github.com/darkautism/pio-lgt8fx/blob/7d5d21c74e5cd3d7734ef398d8b3c074333c718e/builder/frameworks/arduino.py#L40-L46

However the framework code also wants `F_OSC` defined (frequency of the oscillator / raw clock before division) in order to calculate the divisor to produce F_CPU.

https://github.com/dbuezas/lgt8fx/blob/e844e1c88a5d6dbef352ca13e38aacf561450f5d/lgt8f/cores/lgt8f/main.cpp#L146-L172

If this macro is not set, the previous clock divider settings (e.g. set by the bootloader) survive and the code thinks it's e.g. running at 32MHz when in reality the clock divider was never set by the framework and it's 16MHz in reality.

I'd propose to add a `f_osc` attribute to all board JSON definitions that would be carried over as `F_OSC` in the `env.["CPPDEFINES"]` so that it can be made sure the desired CPU frequency is actually reached.

**Comment by maxgerhardt** [comment]
And FYI, https://github.com/darkautism/pio-lgt8fx/commit/39d466a28eda46d5c966b023cecd10547520ef72 is incomplete because it does not transport the values to the CPPDEFINES in the arduino.py builder script.

That would need something like
```py
# fall back to CPU frequency if not given to achieve a divisor of one
board_f_osc = board.get("build.f_osc", "build.f_cpu")
CPPDEFINES = [ 
    ("F_CPU", "$BOARD_F_CPU"), 
    ("F_OSC", board_f_osc),
    ("CLOCK_SOURCE", board_clock_source), 
    "ARDUINO_ARCH_AVR", 
    "ARDUINO_ARCH_LGT", 
    ("ARDUINO", 10812) 
] 
```

**Comment by TheRealTurtler** [comment]
Any chance this will eventually get added to the default version in PlatformIO?

**Comment by maxgerhardt** [comment]
I see this is fixed per 674425875941653a796a831a17549d38ea62dc79.

**Comment by maxgerhardt** [comment]
>Any chance this will eventually get added to the default version in PlatformIO?

If you follow the install instructions at https://github.com/darkautism/pio-lgt8fx#usage, i.e. clone the repo into the PlatformIO folder, you already get this fix. Everyone starting from Sept 22 got this fix. You can delete the old version and redownload the new one to get the update if needed.

**Comment by TheRealTurtler** [comment]
I just tried re-downloading via PlatformIO, but this fix is not included.
![grafik](https://github.com/darkautism/pio-lgt8fx/assets/16053411/ffe6aa3c-e924-4e2c-97d5-c799cfd727cd)

After manually updating to the F_OSC branch, everything works as expected.

**Comment by maxgerhardt** [comment]
You're right, this is only in the F_OSC branch. It's @darkautism responsibility to merge this back into master.

**Comment by darkautism** [comment]
Should been fixed in framework v1.0.2

![image](https://github.com/darkautism/pio-lgt8fx/assets/3898040/47209948-8d4d-4f8f-bc2e-69ab6779dffc)

Let me know if it's not work.

**Comment by TheRealTurtler** [comment]
Works perfectly now, thanks for the quick update 👍
